src/main/java/mil/dds/anet/utils/Utils.java

```diff
@@ -323,27 +323,79 @@ public static boolean isEmptyOrNull(Collection<?> c) {
    * @return Whether the email is whitelisted
    */
-  public static boolean isEmailWhitelisted(String email, List<String> whitelistDomainNames) {
-    if (isEmptyOrNull(email) || isEmptyOrNull(whitelistDomainNames)) {
+  public static boolean isEmailWhitelisted(final String email,
+      final List<String> whitelistDomainNames) {
+    try {
+      return isLogonDomainInList(email, whitelistDomainNames);
+    } catch (Exception e) {
+      logger.error("Failed to process email: ", email);
       return false;
     }
+  }
+
+  /**
+   * Checks whether an domain user name is allowed according to a list of whitelisted domains.
+   * 
+   * More info: https://docs.microsoft.com/en-us/windows/win32/secauthn/user-name-formats
+   * 
+   * @param email The domain user name to check
+   * @param whitelistDomainNames The list of ignaored domain user names (wildcards allowed)
+   * @return Whether the domain user name is ignored
+   */
+  public static boolean isDomainUserNameIgnored(final String domainUserName,
+      final List<String> ignoredDomainNames) {
+    try {
+      return isLogonDomainInList(domainUserName, ignoredDomainNames);
+    } catch (Exception e) {
+      logger.error("Failed to process domain user name: ", domainUserName);
+      return true;
+    }
+  }
+
+  private static boolean isLogonDomainInList(final String logon, final List<String> list)
+      throws Exception {
 
     final String wildcard = "*";
-    final String[] splittedEmail = email.split("@");
-    final String from = splittedEmail[0].trim();
-    final String domainName = splittedEmail[1].toLowerCase();
 
-    final List<String> wildcardDomainNames = whitelistDomainNames.stream()
-        .filter(domain -> String.valueOf(domain.charAt(0)).equals(wildcard))
-        .collect((Collectors.toList()));
+    // Separator for 'User Principal Name' format
+    final String upnSeparator = "@";
+
+    // Separator for 'Down-Level Logon Name' format
+    final String dlSeparator = "\\";
 
-    final boolean isWhitelistedEmail =
-        from.length() > 0 && whitelistDomainNames.indexOf(domainName) >= 0;
-    final boolean isValidWildcardDomain =
-        wildcardDomainNames.stream().anyMatch(wildcardDomain -> domainName.charAt(0) != '.'
-            && domainName.endsWith(wildcardDomain.substring(1)));
+    if (isEmptyOrNull(logon) || isEmptyOrNull(list)) {
+      return false;
+    }
+
+    // Logon has no domain
+    if (!logon.contains(upnSeparator) && !logon.contains(dlSeparator)) {
+      return false;
+    }
+
+    // Find out the format
+    boolean isDownLevelFormat = logon.contains(dlSeparator);
+
+    final String[] splittedLogon =
+        logon.trim().split(isDownLevelFormat ? dlSeparator + "\\" : upnSeparator);
+
+    // A logon (domain user name or email) is expected to have two parts: username<separator>domain
+    if (splittedLogon.length != 2) {
+      throw new Exception("Misformed logon: " + logon);
```

Comment:
Please use a more specific exception (see above); "Misformed" should be "Malformed".

Reply:
My grammar is malformed :P